lib/controllers/git-tab-controller.js

```diff
@@ -327,16 +333,25 @@ export default class GitTabController {
     return this.getActiveRepository().checkout(branchName, options);
   }
 
-  focus() {
-    this.refs.gitTab.focus();
+  @autobind
+  rememberLastFocus(event) {
+    this.lastFocus = this.refs.gitTab.rememberFocus(event) || GitTabView.focus.STAGING;
   }
 
-  focusAndSelectStagingItem(filePath, stagingStatus) {
-    return this.refs.gitTab.focusAndSelectStagingItem(filePath, stagingStatus);
+  restoreFocus() {
+    this.refs.gitTab.setFocus(this.lastFocus);
+  }
+
+  hasFocus() {
+    return this.element.contains(document.activeElement);
```

Comment:
It turns out that `.contains()` is [inclusive](https://developer.mozilla.org/en-US/docs/Web/API/Node/contains):

```javascript
> document.body.contains(document.body)
true
```

So, we can save a little typing. Although maybe it's worth it to save an unnecessary DOM operation... ?